### ISSUE TITLE
Anonymous user can create a study!?

### ISSUE THREAD

**Issue by jimallman** [OpenTreeOfLife/opentree#760]
See https://github.com/OpenTreeOfLife/phylesystem-1/commit/52f7a0020cc4814a0f1a1bfb012205511789000c

Assuming this was submitted via the curation tool, it should not be allowed unless the user is logged in.

Possibly a stale session (curator tool open and idle, then saved after session timed out)? Or maybe we don’t guard the create tool as carefully?

**Comment by jimallman** [comment]
As @jar398 points out, it could also be the result of a direct API call. But in that case, would these commits be attributed to user **opentreeapi**?

**Comment by mtholder** [comment]
See: https://github.com/OpenTreeOfLife/opentree/blob/master/webapp/modules/opentreewebapputil.py#L133-L140

**Comment by mtholder** [comment]
In answer to your question, @jimallman: yes the merge and push would still be associated with the opentreeapi user, I think.  We do have some legit curation activity that comes in with ANONYMOUS<ANONYMOUS> as the name and author. See https://github.com/OpenTreeOfLife/phylesystem-1/commit/fdae7a830d3e7338204eb3ded45ac3eb7af7ef70 for example (from just this weekend).

**Comment by kcranston** [comment]
How does that happen? (that = legit activity from anonymous)

**Comment by mtholder** [comment]
I think that it is the code that I highlighted above (and no rejection for ANONYMOUS on the server side). I think that if you fill in nothing on your GH profile, this may be what happens.

**Comment by jimallman** [comment]
The odd thing is, I didn't think you could *have* a session without at least a GitHub login (userid, e.g. 'jimallman'). That's why I'm wondering if this is an edge case where the curation tool is opened by a logged-in user, then their session times out, then they Save changes (and oops! they're no longer logged in). In this case, we should check and block the save, but where would that leave the curator?

We could get the curator's userid from the client and decide to trust it, or we can probably find a way to "refresh" their session through OAuth and carry on.

**Comment by snacktavish** [comment]
So the trick is if you have logged into github, but haven't verified your email, you can commit to phylesystem but the commit will be anonymous. It is a weird loophole- you are logged in, but can't do anything on github unless you have verified your email, but we aren't checking, and then GitHub gets confused when displaying the commit. 
e.g. if snarktavish makes a new account, s/he is logged in, and can therefore save edits on phylesystem, but they show up as anonymous because s/he hasn't verified hir email (which happens to be a hotmail address with my name on it).
![screenshot from 2015-09-22 15 03 52](https://cloud.githubusercontent.com/assets/1652782/10028579/3e720f4e-613b-11e5-99e1-9ab4ca403dfd.png)

Not good behavior, will try to dig into what the fix should be.

**Comment by jimallman** [comment]
Thanks @snacktavish! At least we know what we're up against.

>...unless you have verified your email, but we aren't checking...

I'm not sure we **can** determine this, checking the GitHub User API now...

**Comment by jimallman** [comment]
@snacktavish, if I understand correctly, the email **verification** is distinct from choosing to make your email **public**, yes? 

I was hoping we could make a determination based on whether we see an email address in the [user info returned by the GitHub API](https://developer.github.com/v3/users/#get-the-authenticated-user), but it sounds like that could be misleading.

**Comment by snacktavish** [comment]
Hmm, seems more complicated than I thought! 
It isn't the email verification, even with a verified public email snarktavish's commits show up as ANONYMOUS, even though I can post issues on github using that account and they show up normally (e.g. https://github.com/snacktavish/2013-08-22-ku/issues/4)

**Comment by snacktavish** [comment]
Nope- I was wrong in my previous comment - if it is verified and email is public it behaves correctly.  
As far as I can tell, is there is no public email associated with the account, even if there is a verified github email, commits will show up as anonymous. Email verification is separate, and both need to happen. I suppose that email will only be returned by the github API if it is public though, so that would be a reasonable step. 
Makes the login procedure a bit annoying and multi-step unfortunately, because public email is not the default.

**Comment by jimallman** [comment]
If we can detect the lack of verification, it's not so bad, just another obstacle for the drive-by contributor. We'd need to pester them to confirm email with GitHub before doing any significant work on the site.

(I wonder what happens if they sign up and start curating a study immediately, then verify email **before saving** their work. Would it save properly?)
